pkg/kube/kube.go

```diff
@@ -588,11 +591,15 @@ func (m *kubePackage) kubePeek(ctx context.Context, url string) (obj runtime.Obj
 	return obj, true, nil
 }
 
+func errImmutableRessource(resource string) error {
+	return fmt.Errorf("cannot update %s. Use -force to delete and recreate", resource)
+}
+
 // mergeObjects merges the fields from the live object to the new
 // object such as resource version and clusterIP.
 // TODO(jon.yucel): Instead of selectively picking fields, holisticly
 // solving this problem requires three-way merge implementation.
-func mergeObjects(live, obj runtime.Object) error {
+func mergeObjects(live, obj runtime.Object) (recreate bool, err error) {
```

Comment:
I wrapped the error, removed the bool since we don't need it and passed on `runtime.Object` to give a more detailed error message. I couldn't return the bool instead of the error itself, cause outside of `mergeObjects()` we don't know which immutable attribute was attempted to change (since it's not the entire resource that's immutable, it's single attributes of a resource that are)